src/Symfony/Component/HttpKernel/Fragment/AbstractSurrogateFragmentRenderer.php

```diff
@@ -64,6 +64,10 @@ public function __construct(SurrogateInterface $surrogate = null, FragmentRender
     public function render($uri, Request $request, array $options = array())
     {
         if (!$this->surrogate || !$this->surrogate->hasSurrogateCapability($request)) {
+            if ($uri instanceof ControllerReference) {
+                $this->checkNonScalar($uri->attributes);
```

Comment:
Instead of making `checkNonScalar()` protected in order to call it here, we could generate the uri instead:

```php
if ($uri instanceof ControllerReference) {
    $uri = $this->generateFragmentUri($uri, $request, false, true);
}

return $this->inlineStrategy->render($uri, $request, $options);
```

This way, however, `_locale` and `_format` might not be passed properly as [the following code in the InlineFragmentRenderer](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/HttpKernel/Fragment/InlineFragmentRenderer.php#L65-L69) wouldn't be called:

```php
foreach (array('_format', '_locale') as $key) {
    if (isset($attributes[$key])) {
        $reference->attributes[$key] = $attributes[$key];
    }
}
```